tensornetwork/backends/abstract_backend.py

```diff
@@ -706,6 +706,23 @@ def inv(self, matrix: Tensor) -> Tensor:
     raise NotImplementedError("Backend '{}' has not implemented `inv`.".format(
         self.name))
 
+  def pinv(self,
+           matrix: Tensor,
+           rcond: float = 1E-15,
+           hermitian: bool = False) -> Tensor:
```

Comment:
I think this function should be given a "pivot axis" argument, similarly to qr - check that function for details. The idea is that a higher-rank tensor would be matricized.

Reply:
Understood ! I've added the pivot axis argument similar to qr.